crates/telio-lana/src/lib.rs

```diff
@@ -115,3 +132,207 @@ pub fn deinit_lana() -> Result<moose::Result, moose::Error> {
+/// * app_version - Indicates the semantic version of the application.
+/// * prod - whether the events should be sent to production or not
+/// * tx - channel tx half for the InitCallback instance
+pub fn init_moose(
+    event_path: String,
+    app_version: String,
+    prod: bool,
+    tx: SyncSender<bool>,
+) -> std::result::Result<moose::Result, moose::Error> {
+    cfg_if! {
+        if #[cfg(test)] {
+            if let Some(mock) = test::MOCK.lock().unwrap().as_ref() {
+                return mock.init(
+                    event_path,
+                    LANA_APP_NAME.to_string(),
+                    app_version,
+                    LANA_MOOSE_VERSION.to_string(),
+                    prod,
+                    Box::new(moose_callbacks::MooseInitCallback {
+                        init_success_tx: tx,
+                    }),
+                    Box::new(moose_callbacks::MooseErrorCallback),
+                )
+            }
+        }
+    }
+    moose::init(
+        event_path,
+        LANA_APP_NAME.to_string(),
+        app_version,
+        LANA_MOOSE_VERSION.to_string(),
+        prod,
+        Box::new(moose_callbacks::MooseInitCallback {
+            init_success_tx: tx,
+        }),
+        Box::new(moose_callbacks::MooseErrorCallback),
+    )
+}
+#[cfg(test)]
+mod test {
+    use serial_test::serial;
+    use std::sync::Mutex;
+
+    use super::{
+        deinit_lana,
+        event_log::moose,
+        init_lana, is_lana_initialized,
+        moose::{ErrorCallback, InitCallback},
+    };
+
+    pub static MOCK: Mutex<Option<MooseStub>> = Mutex::new(None);
+
+    pub struct MooseStub {
+        return_success: bool,
+        init_cb_result: Result<moose::TrackerState, moose::MooseError>,
+        should_call_init_cb: bool,
+    }
+
+    impl MooseStub {
+        #[allow(clippy::too_many_arguments)]
+        pub fn init(
+            &self,
+            _event_path: String,
+            _app_name: String,
+            _app_version: String,
+            _exp_moose_ver: String,
+            _prod: bool,
+            init_cb: Box<(dyn InitCallback + 'static)>,
+            _error_cb: Box<(dyn ErrorCallback + Sync + std::marker::Send + 'static)>,
+        ) -> std::result::Result<moose::Result, moose::Error> {
+            if self.should_call_init_cb {
+                init_cb.after_init(&self.init_cb_result);
+            }
+            if self.return_success {
+                Ok(moose::Result::Success)
+            } else {
+                Err(moose::Error::NotInitiatedError)
+            }
+        }
+    }
+
+    impl Default for MooseStub {
+        fn default() -> Self {
+            MooseStub {
+                return_success: true,
+                init_cb_result: Ok(moose::TrackerState::Ready),
+                should_call_init_cb: true,
+            }
+        }
+    }
+
+    fn set_custom_mock(mock: Option<MooseStub>) {
+        *MOCK.try_lock().unwrap() = mock;
+    }
+
+    fn teardown() {
```

Comment:
This approach doesn't handle exiting via panic, please handle it also.

Reply:
Added wrapper to drop global on panic.

Reply:
Please also use it in all tests that require stub customization and remove all explicit calls to the `teardown` function.

Reply:
We should still call `teardown` in the end of tests that do not need the stub, no?